.travis.yml

```diff
@@ -1,13 +1,41 @@
 # copyright © 2017 the libfulltext authors (see AUTHORS.md and LICENSE)
 
+sudo: false
 language: python
+cache: pip
 python:
-    - "3.5"
-    - "3.6"
-before_install:
-    - pip install -r requirements.txt
-    - pip install pylint nose2
-script:
-    - pylint bin libfulltext
-    - nose2
-    - pip install ${TRAVIS_BUILD_DIR}
+  - "3.5"
+  - "3.6"
+
+install:
+  - travis_retry pip install -r requirements.txt
+  - travis_retry pip install nose2 pylint flake8
+
+script:  # This is the 'test' build stage
+  - nose2 --with-coverage
+    # Fail if coverage below threshold
+  - python3 -m coverage report --fail-under=55 > /dev/null
+  - pip install ${TRAVIS_BUILD_DIR}
+
+after_success:
```

Comment:
"upload coverage" do you still want to do this within this PR? What's your plan here, can we just link on a coverage badge on travis / do you want to upload a browsable html coverage summary?

Reply:
If @andrenarchy does the relevant setup it is not a big deal. I have used [coveralls](https://coveralls.io/) for one of my projects, but I always wanted to give [codecov](https://codecov.io/) a chance. Both seem to have python support ready and the relevant tooling, so in both cases it would be one line of code.